### DISCUSSION TITLE
can't find ReportController.cs in new version  of serenity

### DISCUSSION THREAD

**Discussion by abualmazen** [serenity-is/Serenity#6517] [Help]
Hi, i download new vertion of serenty and i create new project
i got this error
InvalidOperationException: Generated PDF file is empty!
Serenity.Reporting.HtmlToPdfConverter.Execute() in HtmlToPdfConverter.cs
Serenity.Extensions.Pages.ReportController.RenderAsPdf(IReport report, string key, string opt) in ReportController.cs
Serenity.Extensions.Pages.ReportController.Execute(string key, string opt, string ext, bool download, bool printing) in ReportController.cs
i try to find it but can't find ReportController.cs in new virtion of serenity and the folder reporting

**Comment by wezmag** [comment]
This is the `ReportController.cs` you were looking for.
https://github.com/serenity-is/common-features/blob/master/src/Serenity.Extensions/Serenity.Extensions/Reporting/ReportController.cs
